pkg/diag/validator/pod.go

```diff
@@ -71,27 +74,36 @@ var (
 
 // PodValidator implements the Validator interface for Pods
 type PodValidator struct {
-	k          kubernetes.Interface
-	controller metav1.Object
-	recos      []Recommender
+	k      kubernetes.Interface
+	depObj appsv1.Deployment
```

Comment:
is there a more generic type than `appsv1.Deployment` that we can use here. Thinking if we extend the status-check to `statefulsets` in the future, is there a way this code wouldn't need to be refactored again.

Reply:
We could refactor this later. Right now I do want to get this in since we don't do any resource check for statefulsets.